glean-core/src/util.rs

```diff
@@ -48,6 +48,39 @@ pub fn get_iso_time_string(datetime: DateTime<FixedOffset>, truncate_to: TimeUni
 ///
 /// This converts from the `Local` timezone into its fixed-offset equivalent.
 pub(crate) fn local_now_with_offset() -> DateTime<FixedOffset> {
+    #[cfg(target_os = "windows")]
+    {
+        // `Local::now` takes the user's timezone offset
+        // and panics if it's not within a range of [-24, +24] hours.
+        // This causes crashes in a small number of clients on Windows.
+        //
+        // We can't determine the faulty clients
+        // or the circumstancens under which this happens,
+        // so the best we can do is have a workaround:
+        //
+        // We try getting the time and timezone first,
+        // then manually check that it is a valid timezone offset.
+        // If it is, we proceed and use that time and offset.
+        // If it isn't we fallback to UTC.
+        //
+        // This has the small downside that it will use 2 calls to get the time,
+        // but only on Windows.
+        //
+        // See https://bugzilla.mozilla.org/show_bug.cgi?id=1611770.
+
+        use chrono::Utc;
+
+        // Get timespec, including the user's timezone.
+        let tm = time::now();
+        // Same as chrono:
+        // https://docs.rs/chrono/0.4.10/src/chrono/offset/local.rs.html#37
+        let offset = tm.tm_utcoff;
+        if let None = FixedOffset::east_opt(offset) {
+            let now: DateTime<Utc> = Utc::now();
```

Comment:
Instrument it?

I agree, testing it would be nightmarish. Platform-specific, and somehow mocking `utcoff`... ugh. But with instrumentation we could at least see if we see the same number of folks with artificial UTC as the number we previously saw crashing.

Reply:
Uff, I would like to instrument this, but it turns out instrumenting internals that are used in the init part of Glean is ... hard.